widgetssdk/src/main/java/com/glia/widgets/chat/ChatManager.kt

```diff
@@ -195,6 +197,12 @@ internal class ChatManager(
             appendNewChatMessageUseCase(messagesState, chatMessage)
             if (chatMessage.chatMessage is VisitorMessage) {
                 checkUnsentMessages(messagesState)
+            } else {
+                hasOngoingSecureConversationUseCase {
+                    if (it && !isQueueingOrLiveEngagementUseCase.hasOngoingLiveEngagement) {
+                        markMessagesRead()
+                    }
```

Comment:
I'd prefer here to check only for `unread_messages_count > 0` because even if there is ongoing live engagement, it means that the visitor 100% saw the messages.